### DISCUSSION TITLE
Surplus E files after executing s3_decomp.i

### DISCUSSION THREAD

**Discussion by rl3fz** [idaholab/moose#18076] [Q&A Modules: Phase field] [answered]
1) Hello, I've executed the 's3_decomp.i' file for spinodal decomposition (Step 3 of the phase field tutorial), but in addition to the corresponding file (of type E), 82 other files (of type E-S002, E-S003, E-S004, etc.) were created.

This didn't happen in previous steps, and by process of elimination the added files seem to be a result of the changes to the mesh and executioner blocks for adaptive meshing.

Is this supposed to happen? I don't see any difference in mesh quality between the E-S002 file and the E-S083 files, so I'm not sure why they would be necessary in the first place. If this shouldn't happen, advice on preventing it would be greatly appreciated.

2) I'm also not sure whether I should be using the [./elapsed] or [./active_time] postprocessor to determine the time spent on the simulation. They both seem to work fine, but if one is outdated or has another use, I would appreciate any info here too (I couldn't find documentation for either).

**Comment by GregVernon** [comment]
Exodus doesn't (yet) support dynamic mesh topology within a single Exodus file, instead they create a new "state" Exodus file (the `e-sXXX` files) if the mesh connectivity (topology) changes.  Dynamic mesh topology is often caused by using mesh-adaptivity.  ParaView is able to recognize these "extra" Exodus files as being sequential so this shouldn't affect visualization in ParaView, if that's a concern.

You can find out more about the dynamic topology and Exodus from [the SEACAS project page here](https://github.com/gsjaardema/seacas/wiki/Dynamic_Topology).

**Reply by rl3fz** [reply]
Thank you!

**Comment by ngrilli** [comment, marked as answer]
Dear @rl3fz 

Yes, it's normal.
That happens also if you use element elimination methods that modify the mesh,
for instance the one described here:
https://arxiv.org/abs/2105.13257

and the test cases here:
https://github.com/idaholab/moose/tree/next/test/tests/userobjects/element_subdomain_modifier

If you open the main .e file in paraview, the one without -s*, 
all the time steps will be automatically accessible in paraview.
Just don't delete the -s* files .

Best Regards,
Nicolò

**Reply by rl3fz** [reply]
Thank you!

**Comment by GiudGiud** [comment]
Wrt to the postprocessors you should use, the PerfGraphData should have all you need for timing your application. 
Note that [./elapsed] or [./active_time] are just the name of the objects, they can be set to anything. What matters for you is the type of the object

**Reply by rl3fz** [reply]
Thank you!
